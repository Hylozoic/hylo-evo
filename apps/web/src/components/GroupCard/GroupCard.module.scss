.card {
  border: 1px solid $color-cape-cod-10;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  right: 0px;
  transition: 0.25s;
  display: flex;
  flex-direction: column;
  &:hover {
    box-shadow: 0px 0px 6px 0px $color-black-10;
    text-decoration: none;
  }
}

.group-description {
  @include font-regular;
  padding: 15px 15px 0 15px;
  width: 100%;
  font-size: 12px;
  line-height: 1.5;
  color: $color-regent;
  opacity: 0.8;
  max-height: 4.5em;
  margin-bottom: 15px;
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
}

.group-tags {
  max-height: 68px;
  overflow: hidden;
  padding: 0 15px 0 15px;
  margin-bottom: 15px;
}

.tag-pill{
  font-family: Circular Std;
  text-transform: none;
  font-style: normal;
  font-weight: 400;
  font-size: 11px;
  line-height: 18px;
}

.constrained.card {
  border-top: 5px solid $color-cape-cod-10;
  padding-top: 10px;
}

.expanded,
.expanded:hover {
  box-shadow: 0px 4px 6px 0px $color-black-10;
}

.group-link {
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }
}
