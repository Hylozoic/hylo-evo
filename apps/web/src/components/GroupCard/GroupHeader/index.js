import component from './GroupHeader'

export default component
