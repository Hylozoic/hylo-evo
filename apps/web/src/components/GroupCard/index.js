import component from './GroupCard'

export default component
