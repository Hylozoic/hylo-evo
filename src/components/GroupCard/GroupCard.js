import React from 'react'
import { Link } from 'react-router-dom'
import { TextHelpers } from 'hylo-shared'
import { groupUrl, groupDetailUrl } from 'util/navigation'
import ClickCatcher from 'components/ClickCatcher'
import GroupHeader from './GroupHeader'
import HyloHTML from 'components/HyloHTML'
import cx from 'classnames'
import './GroupCard.scss'

/*
  Each card needs
  - group title
  - group icon
  - group background (tiny, behind the group icon)
  - member sample (for avatars on right )
  - group geography descriptor (indigenous territory, location)

  TODO: Then is contents changed based on group type... perhaps passed in as a Content component
*/

export default function GroupCard ({
  memberships,
  group = {},
  routeParams = {},
  highlightProps = {},
  className,
  expanded = false,
  constrained = false,
  onClick = () => {}
}) {
  // XXX: turning this off for now because topics are random and can be weird. Turn back on when groups have their own #tags
  // const topics = group.groupTopics && group.groupTopics.toModelArray()

  const linkTo = memberships.includes(group.id)
    ? groupUrl(group.slug)
    : groupDetailUrl(group.slug, routeParams)

  return (
    <Link to={linkTo} styleName='group-link'>
      <div
        onClick={onClick}
        styleName={cx('card', { expanded }, { constrained })}
        className={className}>
        <GroupHeader
          {...group}
          group={group}
          routeParams={routeParams}
          highlightProps={highlightProps}
          constrained={constrained}
        />
        {group.description
          ? <div styleName='group-description'>
            <ClickCatcher>
              <HyloHTML element='span' html={TextHelpers.markdown(group.description)} />
            </ClickCatcher>
          </div>
          : ''
        }

        {/* XXX: turning this off for now because topics are random and can be weird. Turn back on when groups have their own #tags
         {topics.length > 0
           ? <div styleName='group-tags'>
             {topics.map((topic, index) => (
               <Pill
                 styleName='tag-pill'
                 darkText
                 label={capitalize(topic.topic.name.toLowerCase())}
                 id={topic.id}
                 key={index}
               />
             ))}
           </div>
           : ''
         } */}
      </div>
    </Link>
  )
}
