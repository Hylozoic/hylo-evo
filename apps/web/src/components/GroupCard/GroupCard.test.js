import GroupCard from './GroupCard'
import { shallow } from 'enzyme'
import React from 'react'
const props = {
  group: {
    id: 1,
    name: 'A Great Cause',
    slug: 'great-cause',
    groupTopics: {
      toModelArray: () => [
        { topic: {
          name: 'Life',
          id: '1wooooop'
        } },
        { topic: {
          name: 'Love',
          id: '2wooooop'
        } },
        { topic: {
          name: 'Light',
          id: '3wooooop'
        } },
        { topic: {
          name: 'LOLS',
          id: '4wooooop'
        } }
      ]
    },
    description: 'the description ' +
      'the description ' +
      'the description ' +
      'the description ' +
      'the description '
  },
  expanded: false,
  memberships: [12, 24, 25, 346]
}

it('matches last snapshot', () => {
  const wrapper = shallow(<GroupCard {...props} />)
  expect(wrapper).toMatchSnapshot()
})

it('matches last snapshot with different config', () => {
  const differentProps = {
    ...props,
    constrained: true
  }
  const wrapper = shallow(<GroupCard {...differentProps} />)
  expect(wrapper).toMatchSnapshot()
})
