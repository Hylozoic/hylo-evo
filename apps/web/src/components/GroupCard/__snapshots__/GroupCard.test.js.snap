// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches last snapshot 1`] = `
<Link
  data-stylename="group-link"
  to="/all/group/great-cause"
>
  <div
    data-stylename="card"
    onClick={[Function]}
  >
    <GroupHeader
      constrained={false}
      description="the description the description the description the description the description "
      group={
        Object {
          "description": "the description the description the description the description the description ",
          "groupTopics": Object {
            "toModelArray": [Function],
          },
          "id": 1,
          "name": "A Great Cause",
          "slug": "great-cause",
        }
      }
      groupTopics={
        Object {
          "toModelArray": [Function],
        }
      }
      highlightProps={Object {}}
      id={1}
      name="A Great Cause"
      routeParams={Object {}}
      slug="great-cause"
    />
    <div
      data-stylename="group-description"
    >
      <ClickCatcher>
        <HyloHTML
          element="span"
          html="<p>the description the description the description the description the description </p>
"
        />
      </ClickCatcher>
    </div>
  </div>
</Link>
`;

exports[`matches last snapshot with different config 1`] = `
<Link
  data-stylename="group-link"
  to="/all/group/great-cause"
>
  <div
    data-stylename="card constrained"
    onClick={[Function]}
  >
    <GroupHeader
      constrained={true}
      description="the description the description the description the description the description "
      group={
        Object {
          "description": "the description the description the description the description the description ",
          "groupTopics": Object {
            "toModelArray": [Function],
          },
          "id": 1,
          "name": "A Great Cause",
          "slug": "great-cause",
        }
      }
      groupTopics={
        Object {
          "toModelArray": [Function],
        }
      }
      highlightProps={Object {}}
      id={1}
      name="A Great Cause"
      routeParams={Object {}}
      slug="great-cause"
    />
    <div
      data-stylename="group-description"
    >
      <ClickCatcher>
        <HyloHTML
          element="span"
          html="<p>the description the description the description the description the description </p>
"
        />
      </ClickCatcher>
    </div>
  </div>
</Link>
`;
